src/pages/Page404/Page404.tsx

```diff
@@ -0,0 +1,28 @@
+import { LayoutPage } from 'components/LayoutPage/LayoutPage';
+import { useEffect, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { AppRoutes } from 'types/AppRoutes';
+
+export function Page404() {
+  const [time, setTime] = useState(5);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (!time) {
+      navigate(AppRoutes.index);
+    }
+
+    const interval = setInterval(() => setTime(time - 1), 1000);
+
+    return function () {
+      clearInterval(interval);
+    };
+  }, [time]);
```

Comment:
Это ни на что не повлияет, но похоже navigate должен быть в зависимости. Но тогда это нужно разделить это на два useEffect. В общем я не уверен) дождемся Антона.
```suggestion
   useEffect(() => {
    if (!time) {
      navigate(AppRoutes.index);
    }
  }, [time, navigate]);

  useEffect(() => {
    const interval = setInterval(() => setTime(time - 1), 1000);

    return function () {
      clearInterval(interval);
    };
  }, [time]);
```

Reply:
Нужен, но делить на 2 эффекта не надо, он довольно ёмкий и всё в нём по делу)

Reply:
Если с navigate что случится, лишний запуск интервала получим

Reply:
Если с ним что-то случится, то это хороший повод разобраться, потому что в таком случае лишний интервал будет меньшей из проблем

Reply:
Эм, useEffect не лучшее место для дебага)

Reply:
Я говорю не про дебаг, а про то, что ситуация с изменением navigate мифическая, он изменится лишь в том случае, если сменится `pathname`, то есть когда мы уйдём со страницы. На самом деле [не только](https://github.com/remix-run/react-router/blob/main/packages/react-router/index.tsx) в этом случае, но остальные нас не касаются
Поэтому в данном случае нет причин забивать на rules of hooks

Reply:
Не согласен. Мифические кейсы приводят к реальным, а эффект, который должен обновляться исключительно от изменения time имеет лишнюю зависимость. Потом появится ещё одна. Потенциальная ошибка.

Reply:
Так это не забитее на правила хуков. Нужно просто разбить хук на две части

Reply:
Сейчас проблем не будет, а когда будет добавляться что-то новое, тогда уже можно будет подумать про распил. Это уже не про мифологию, а поддержку кода)
Но разбить можно, будет норм. Это даже не будет оверхедом, но конкретно в этом случае ты излишне запарился

Reply:
Я просто так в своё время настрадался с эффектами и ререндерами, что трегирит по любому поводу)